### ISSUE TITLE
Crash during SDK initialization in OPPO Smartphones

### ISSUE THREAD

**Issue by jfsso** [intercom/intercom-android#263]
## Version info
  - Intercom for Android version: 3.0.9
  - Android OS version: 4.x & 5.x
  - Android Device: Any OPPO device

## Expected behavior
SDK initializes without crashing

## Actual behavior
Crashes during SDK initialization when trying to start IntercomGcmRegistrationService.

The service start fails with a SecurityException because of a battery saving setting in OPPO devices. Would be great if the SDK could catch this and fail gracefully.

## Steps to reproduce
 1. Call Intercom#initialize on an OPPO device

## Logs

```
Fatal Exception: java.lang.RuntimeException: Unable to create application com.anprosit.drivemode.DriveModeApplication: java.lang.SecurityException: Unable to start service Intent { cmp=com.drivemode.android/io.intercom.android.sdk.gcm.IntercomGcmRegistrationService }: Unable to launch app com.drivemode.android/10133 for service Intent { cmp=com.drivemode.android/io.intercom.android.sdk.gcm.IntercomGcmRegistrationService }: user 0 is restricted
       at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4783)
       at android.app.ActivityThread.access$1700(ActivityThread.java:164)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1523)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:159)
       at android.app.ActivityThread.main(ActivityThread.java:5541)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:964)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:759)
Caused by java.lang.SecurityException: Unable to start service Intent { cmp=com.drivemode.android/io.intercom.android.sdk.gcm.IntercomGcmRegistrationService }: Unable to launch app com.drivemode.android/10133 for service Intent { cmp=com.drivemode.android/io.intercom.android.sdk.gcm.IntercomGcmRegistrationService }: user 0 is restricted
       at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1768)
       at android.app.ContextImpl.startService(ContextImpl.java:1741)
       at android.content.ContextWrapper.startService(ContextWrapper.java:527)
       at io.intercom.android.sdk.IntercomPushManager.startGcmRegistrationService(SourceFile:31)
       at io.intercom.android.sdk.RealIntercom.(SourceFile:94)
       at io.intercom.android.sdk.Intercom.initialize(SourceFile:43)
       at com.anprosit.drivemode.DriveModeApplication.onCreate(SourceFile:183)
```

**Comment by ronocod** [comment]
Thanks for reporting this, we're working on a fix now.

**Comment by ronocod** [comment]
Fixed in 3.0.10 🎉
